lib/rack/jwt/auth.rb

```diff
@@ -159,20 +164,30 @@ def check_valid_algorithm!
       end
 
       def check_exclude_type!
-        raise ArgumentError, 'exclude argument must be an Array' unless @exclude.is_a?(Array)
+        raise ArgumentError, 'exclude argument must be an Array' unless exclude.is_a?(Array)
 
-        @exclude.each do |x|
-          raise ArgumentError, 'each exclude Array element must be a String' unless x.is_a?(String)
-          raise ArgumentError, 'each exclude Array element must not be empty' if x.empty?
+        exclude.each do |exclusion|
+          raise ArgumentError, 'each exclude Array element must be a Hash' unless exclusion.is_a?(Hash)
+
+          if exclusion.keys.sort != %i(methods path)
+            raise ArgumentError, 'each exclude Array element must have only path and methods as keys'
+          end
 
-          unless x.start_with?('/')
-            raise ArgumentError, 'each exclude Array element must start with a /'
+          unless exclusion[:path].start_with?('/')
+            raise ArgumentError, 'each exclude Array element path value must start with a /'
+          end
+
+          unless exclusion[:methods] == :all || exclusion[:methods].is_a?(Array)
```

Comment:
This isn't backwards compatible. If the `methods` is key is omitted it'll raise an exception.